charms/istio-pilot/src/charm.py

```diff
@@ -773,6 +842,26 @@ def _wait_for_update_rollout(
     return versions
 
 
+def get_routes_from_ingress_interface(
+    ingress_interface: Optional[dict], this_app: Application
+) -> dict:
+    routes = {}
+
+    if ingress_interface:
+        sorted_interface_data = sorted(
+            ingress_interface.get_data().items(), key=lambda tup: tup[0][0].id
+        )
+
+        for (rel, app), route in sorted_interface_data:
+            if app != this_app:
+                routes[(rel, app)] = route
```

Comment:
that block basically means that we pick the latest route and override in some cases ?

what is the case when it will happen ?

Reply:
The ingress_interface.get_data() returns a dict of format:
`(relation, application): route_data`
where application might be either us, or the other related application

We go through that data and pull all the data that is from other applications, ignoring the data that is from us.

I think it is conceptually similar to something like:
```python
for relation in all_relations:
  try:
    routes.pop(relation, self.app)
  except KeyError:
    # ignore
```